src/main/java/de/dennisguse/opentracks/services/handlers/MonotonicClock.java

```diff
@@ -0,0 +1,41 @@
+package de.dennisguse.opentracks.services.handlers;
+
+import android.os.SystemClock;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneId;
+
+/**
+ * A monotonic clock relative to the device's clock at time of the instantiation.
+ * Uses device time since startup.
+ * Replacement for java.time.Clock as may not be monotonic.
+ */
+public class MonotonicClock extends Clock {
+
+    private final long startEpoch;
+
+    private final long elapsedRelatimeAtInstation;
```

Comment:
Did you mean `elapsedRealtimeAtInstantation`?

Reply:
Yes ;)